nan.h

```diff
@@ -1552,10 +1567,62 @@ class Callback {
 #endif
   }
 
+  inline v8::Local<v8::Value>
+  Call(v8::Local<v8::Object> target
+     , int argc
+     , v8::Local<v8::Value> argv[]
+     , AsyncResource* resource) const {
+#if (NODE_MODULE_VERSION >= NODE_8_0_MODULE_VERSION)
+    v8::Isolate* isolate = v8::Isolate::GetCurrent();
+
+    return Call_(isolate, target, argc, argv, resource);
+#elif (NODE_MODULE_VERSION > NODE_0_10_MODULE_VERSION)
+    v8::Isolate *isolate = v8::Isolate::GetCurrent();
+    return Call_(isolate, target, argc, argv);
+#else
+    return Call_(target, argc, argv);
+#endif
+  }
+
+  inline v8::Local<v8::Value>
+  Call(int argc, v8::Local<v8::Value> argv[], AsyncResource* resource) const {
+#if (NODE_MODULE_VERSION >= NODE_8_0_MODULE_VERSION)
+    v8::Isolate* isolate = v8::Isolate::GetCurrent();
+    v8::EscapableHandleScope scope(isolate);
+    return scope.Escape(Call_(isolate
+                            , isolate->GetCurrentContext()->Global()
+                            , argc
+                            , argv
+                            , resource));
+#elif (NODE_MODULE_VERSION > NODE_0_10_MODULE_VERSION)
+    v8::Isolate *isolate = v8::Isolate::GetCurrent();
+    v8::EscapableHandleScope scope(isolate);
+    return scope.Escape(
+        Call_(isolate, isolate->GetCurrentContext()->Global(), argc, argv));
+#else
+    v8::HandleScope scope;
+    return scope.Close(Call_(v8::Context::GetCurrent()->Global(), argc, argv));
+#endif
+  }
+
  private:
   NAN_DISALLOW_ASSIGN_COPY_MOVE(Callback)
   Persistent<v8::Function> handle_;
 
+#if (NODE_MODULE_VERSION >= NODE_8_0_MODULE_VERSION)
+  v8::Local<v8::Value> Call_(v8::Isolate *isolate
+                           , v8::Local<v8::Object> target
+                           , int argc
+                           , v8::Local<v8::Value> argv[]
+                           , AsyncResource* resource) const {
+    EscapableHandleScope scope;
+
+    v8::Local<v8::Function> func = New(handle_);
+    return scope.Escape(resource->runInAsyncScope(target, func, argc, argv)
+                                  .ToLocalChecked());
```

Comment:
Admittedly, I am a little bit unsure here about the `ToLocalChecked`. I think we ought to change the prototype here to return a `MaybeLocal` rather than a `Local`. I need to go figure out `Escape` and `MaybeLocal` are supposed to interact.

Reply:
It is a legacy remnant from the time when `v8::Function::Call` returned a `Local<T>`. I think everything dealing with `v8::Function` works this way. It's been planned to change in the next major release for years, but that release has not had to happen so far.

As for `MaybeLocal` and `Escape`, there is a non-explicit conversion constructor `MaybeLocal(Local<T>)`, so just escape the `Local` and it will be wrapped on return.

Reply:
Thanks! Done.